common.go

```diff
@@ -269,6 +272,86 @@ func getValueFromBase64(b64Value string) ([]byte, error) {
 	return value, nil
 }
 
+// loadCACert returns CA set of certificates (cert pool)
+// reads CA certificate based on the certConfig and adds it to the pool
+func loadCACert(certConfig string) (*x509.CertPool, error) {
+	certificates, err := x509.SystemCertPool()
+	if certificates == nil || err != nil {
+		certificates = x509.NewCertPool()
+	}
+
+	if len(certConfig) > 0 {
+		caCert, err := getCertificate(certConfig)
+		if err != nil {
+			return nil, err
+		}
+
+		certificates.AppendCertsFromPEM(caCert)
+	}
+
+	return certificates, nil
+}
+
+// loadServerCertAndKey reads server certificate and associated key file based on certConfig and keyConfig
+// returns parsed server certificate
+// if the private key is encrypted, it will be decrypted using key file passphrase
+func loadServerCertAndKey(certConfig, keyConfig, keyPassConfig string) ([]tls.Certificate, error) {
+	var certificates []tls.Certificate
+
+	// Read cert file
+	certFileBytes, err := getCertificate(certConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	// Read key file
+	keyFileBytes, err := getCertificate(keyConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	// Decode PEM data
+	keyBlock, _ := pem.Decode(keyFileBytes)
```

Comment:
I'm not familiar with the concept, but are there any multi-part files in which the `rest` from `Decode` would be significant?

Reply:
Yes, there could be multiple PEM blocks in a single file.
For example, if we had a single config to accept both certificate and key in a single file, checking `rest` would be significant. But I think it is not applicable here.